components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1094,196 +1315,376 @@ export function createWalletApi () {
         )
       }),
       getTokenBalancesForChainId: query<
```

Comment:
We should probably follow up and rename this to `getAddressTokenBalancesForChainId` since it is per-address

Reply:
This probably should also be moved to the base cache so we can reuse values within the balance registry query (can be done in a follow-up)